plugins/scenario/hu.bme.mit.gamma.scenario.model/src/hu/bme/mit/gamma/scenario/model/reduction/SimpleScenarioGenerator.java

```diff
@@ -70,16 +81,44 @@ public ScenarioDefinition execute() {
 		simple.setChart(factory.createChart());
 		simple.getChart().setFragment(factory.createInteractionFragment());
 		simple.setInitialblock(handleInitBlockCopy());
+		refResolver.resolveReferences(base);
 		for (Annotation annotation : base.getAnnotation()) {
 			simple.getAnnotation().add((Annotation) this.doSwitch(annotation));
 		}
 		previousFragment = simple.getChart().getFragment();
 		for (Interaction interaction : base.getChart().getFragment().getInteractions()) {
 			simple.getChart().getFragment().getInteractions().add((Interaction) this.doSwitch(interaction));
 		}
+		inLineExpressions(simple, base);
 		return simple;
 	}
 
+	private void inLineExpressions(ScenarioDefinition simple, ScenarioDefinition base) {
+		List<DirectReferenceExpression> references = ecoreUtil.getAllContentsOfType(simple,
+				DirectReferenceExpression.class);
+		for (DirectReferenceExpression direct : references) {
+			Declaration decl = direct.getDeclaration();
+			if (decl instanceof ConstantDeclaration) {
+				ConstantDeclaration _const = (ConstantDeclaration) decl;
+				Expression cloned = ecoreUtil.clone(_const.getExpression());
+				ecoreUtil.change(cloned, direct, direct.eContainer());
```

Comment:
Extract `direct.eContainer()` into a local variable, it can be useful later for debugging.